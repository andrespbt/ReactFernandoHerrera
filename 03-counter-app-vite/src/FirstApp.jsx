import PropTypes from 'prop-types';
export const FirstApp = ({ title, subTitle, name }) => {
	if (!title) {
		throw new Error('El titulo no existe');
	}
	return (
		<>
			<h1 data-testid="test-title">{title}</h1>
			<h2>{subTitle}</h2>
			<h2>{name}</h2>
		</>
	);
};

FirstApp.propTypes = {
	subTitle: PropTypes.number.isRequired,
	title: PropTypes.string.isRequired,
};

// FirstApp.defaultProps = {
// 	subTitle: 12345,
// 	title: 'No hay titulo',
// };
