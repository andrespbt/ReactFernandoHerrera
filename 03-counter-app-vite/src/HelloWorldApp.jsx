export const HelloWorldApp = () => {
  return (
    <h1>Hola Mundo!!</h1>
  )
}

//Tarea
//Crear un componente llamado FirstApp