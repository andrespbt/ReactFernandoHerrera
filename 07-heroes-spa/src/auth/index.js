// eslint-disable-next-line import/export
export * from "./pages";
export * from "./context";
