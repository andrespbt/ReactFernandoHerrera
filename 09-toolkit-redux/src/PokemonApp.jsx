import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getPokemons } from './store/slices/pokemon/thunks';

export const PokemonApp = () => {
  const dispatch = useDispatch();
  const { page, isLoading, pokemons = [] } = useSelector(state => state.pokemons);

  useEffect(() => {
    dispatch(getPokemons());
  }, []);

  return (
    <>
      <h1>Pokemon App</h1>
      <hr />

      <span>Is loading ? {isLoading ? 'true' : 'false'}</span>

      <ul>
        {pokemons.map(pokemon => (
          <li key={pokemon.name}>
            <h3>{pokemon.name}</h3>
          </li>
        ))}
      </ul>

      <button
        disabled={isLoading}
        onClick={() => dispatch(getPokemons(page))}>
        Next
      </button>
    </>
  );
};
