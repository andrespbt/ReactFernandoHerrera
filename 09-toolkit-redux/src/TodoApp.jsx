import { useState } from 'react';
import { useGetTodoByIdQuery, useGetTodosQuery } from './store/apis/todosApi';

export const TodoApp = () => {
  const [todoId, setTodoId] = useState(1);
  //   const { data: todos = [], isLoading } = useGetTodosQuery();
  const { data: todo, isLoading } = useGetTodoByIdQuery(todoId);

  const nextTodo = () => {
    setTodoId(todoId + 1);
  };

  const prevTodo = () => {
    if (todoId > 0) setTodoId(todoId - 1);
  };

  return (
    <>
      <h1>Todo RTK Query</h1>
      <hr />

      <h4>Is Loading {isLoading ? 'true' : 'false'}</h4>

      <pre>{JSON.stringify(todo)}</pre>

      <button onClick={prevTodo}>Prev Todo</button>
      <button onClick={nextTodo}>Next Todo</button>

      {/* <ul>
        {todos.map(({ userId, title, id, completed }) => (
          <li key={id}>
            <strong>{completed ? 'DONE' : 'PENDING'}</strong>
            {title}
          </li>
        ))}
      </ul> */}
    </>
  );
};
