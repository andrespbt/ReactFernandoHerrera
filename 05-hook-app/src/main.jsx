import React from 'react';
import ReactDOM from 'react-dom/client';
import { TodoApp } from './08-useReducer/TodoApp';
// import { Padre } from './07-tarea-memo/Padre';
// import { CallbackHook } from './06-memos/CallbackHook';
// import { MemoHook } from './06-memos/MemoHook';
// import { Memorize } from './06-memos/Memorize';
// import { Layout } from './05-useLayoutEffect/Layout';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { SimpleForm } from './02-useEffect/SimpleForm';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { CounterApp } from './01-useState/CounterApp';
// import { HooksApp } from './HooksApp';
// import './08-useReducer/into-reducer';
import './index.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	// <React.StrictMode>
	<TodoApp />
	// </React.StrictMode>
);
