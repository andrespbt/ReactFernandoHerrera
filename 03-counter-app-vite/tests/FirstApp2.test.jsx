/** @jest-environment jsdom */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas de <FirstApp/>', () => {
	const title = 'Hola, soy Andrés';
	const subTitle = 1994;

	test('Debe hacer match con el snapshot', () => {
		const { container } = render(
			<FirstApp
				title={title}
				subTitle={subTitle}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el mensaje "Hola, soy Andrés"', () => {
		render(
			<FirstApp
				title={title}
				subTitle={subTitle}
			/>
		);

		expect(screen.getByText(title)).toBeTruthy();
	});

	test('Debe mostrar el titulo en un h1', () => {
		render(
			<FirstApp
				title={title}
				subTitle={subTitle}
			/>
		);

		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
	});

	test('Debe mostrar el subtitulo enviado por props', () => {
		render(
			<FirstApp
				title={title}
				subTitle={subTitle}
			/>
		);

		expect(screen.getByText(subTitle)).toBeTruthy();
	});
});
