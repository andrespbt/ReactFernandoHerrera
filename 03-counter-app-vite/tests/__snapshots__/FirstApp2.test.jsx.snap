// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <FirstApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Andr√©s
  </h1>
  <h2>
    1994
  </h2>
  <h2 />
</div>
`;
